engine/src/main/java/org/terasology/logic/players/PlayerSystem.java

```diff
@@ -88,6 +97,15 @@ public void update(float delta) {
                 i.remove();
             }
         }
+
+        i = clientsPreparingToRespawn.iterator();
+        while (i.hasNext()) {
+            SpawningClientInfo spawning = i.next();
+            if (worldProvider.isBlockRelevant(spawning.position)) {
+                respawnPlayer(spawning.clientEntity);
+                i.remove();
```

Comment:
No, removing using the iterator is the safe way to do it. 
https://stackoverflow.com/questions/18448671/how-to-avoid-concurrentmodificationexception-while-removing-elements-from-arr